libraries/AP_Common/Location.cpp

```diff
@@ -293,6 +293,19 @@ Vector2F Location::get_distance_NE_ftype(const Location &loc2) const
                     diff_longitude(loc2.lng,lng) * ftype(LOCATION_SCALING_FACTOR) * longitude_scale((lat+loc2.lat)/2));
 }
 
+// return the distance in meters in North/East/Down plane as a N/E/D vector to loc2, taking into account alt frame correctly
+Vector3f Location::get_distance_NED_alt_frame(const Location &loc2) const
```

Comment:
We do have a `get_alt_distance` now. So you could do a combination of `get_distance_NE` for XY and `get_alt_distance` for Z. That would remove the code duplication.